adr/ADR-40.md

```diff
@@ -125,6 +126,46 @@ all reconnect options are respected.
 For most clients, that means having a `reconnect` method on the
 Client/Connection handle.
 
+#### Custom server pool
+
+Client should have a way to specify custom server pool for reconnection
+attempts. When specified, client should use only those servers for reconnection
+attempts, ignoring any advertised servers from the Server.
+
+New pool should be used for all reconnection attempts until the client is
+restarted or a new pool is specified. New pool is subject to the same rules as
+the default one - randomization unless disabled, max reconnect attempts, etc.
```

Comment:
The pool should not _have to_ care about anything of the normal options or behavior, but it can care if it choses. This is the point of suppling a custom pool, to allow it to do anything it wants.

This means that randomization and the act of resolving ip addresses are part of the pool behavior.

Max reconnect attempts is not part of the pool behavior, it's part of the behavior that uses the pool. The behavior should ask for a resolved ip address, but the pool can chose what the result of a resolved ip address request is, so for instance it could give a resolved address or not when asked for an address

Reply:
ie - once the server list is set, those are the only servers - on connect, the client will learn new servers - if the client doesn't want to use gossiped servers, it should specify its option to ignore cluster updates.